src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -328,6 +329,7 @@ type WarningCode =
     | MissingEntryPoint = 6202
     | IgnoredEntryPoint = 6203
     | ReservedEntryPointArgumentName = 6204
+    | NonResultTypeReturnedInEntryPoint = 6205
```

Comment:
I'm curious, what happens if someone ignores this warning?

Reply:
Our end-to-end setup for QE is still a bit in flux, so I am not sure I know at this point in time. The reason I made it a warning is because it seems like a valid use case that people might want to compile for a certain architecture but only get resource estimations rather than running it on quantum hardware.

Reply:
OK. If it's a valid use case, do we want a way to disable the warning, to tell the compiler that they don't intend to run it on hardware?